IA-2157: Add attachments to forms

bmonjoie opened the pull request:
To embed external media/datasets into ODK forms, Iaso must be able to provide attachment to the Android application and the Enketo server.

Related JIRA tickets : IA-2157

## Self proofreading checklist

- [x] Did I use eslint and black formatters
- [X] Is my code clear enough and well documented
- [X] My migrations file are included
- [X] Are there enough tests
- [ ] Documentation has been included (for new feature)

## Changes

A new `FormAttachment` model has been added with its own suite of APIs.

## How to test

The unit tests are self explainatory IMHO

## Notes

This is the first part as I still need to allow the application to download the manifest with all the attachments

bmonjoie committed:
Add attachments to forms

IA-2157

bmonjoie committed:
Try to please black...

bmonjoie committed:
Try to please MyPy...

olethanh commented on .pre-commit-config.yaml:
```diff
@@ -2,7 +2,7 @@
 repos:
 -   repo: https://github.com/ambv/black
-    rev: 21.5b1
+    rev: 23.3.0
```
this will cause conflict since it's not the same verywhere

bmonjoie replied:
I cannot run black without this because of click. Shouldn't we update black everywhere to fix this issue?

olethanh replied:
do you have the version of click that is in the requirements-dev.txt?

We can change it everywhere if this version is compatible with python 3.8 but I think there was some settings change so it might need modification of a bunch of file

bmonjoie committed:
Add manifest url

Also, reverting `.pre-commit-config.yaml`

IA-2157

bmonjoie committed:
Remove OverwriteStorage as it won't work with S3

bmonjoie committed:
Remove OverwriteStorage as it won't work with S3
Fix documentation
Added `.distinct()` as advised during code review
Added a `unique_together` constraint
Fix tests after removing `OverwriteStorage`

olethanh committed:
Add docstring

olethanh committed:
reorganize manifest action
